codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -214,57 +214,38 @@ let getMostBehindPartitions = (
   ->Js.Array.slice(~start=0, ~end_=maxNumQueries)
 }
 
-let updatePartition = (self: t, ~fetchState: FetchState.t, ~partitionId: partitionIndex) => {
-  {...self, partitions: self.partitions->Utils.Array.setIndexImmutable(partitionId, fetchState)}
-}
-
 type nextQueries = WaitForNewBlock | NextQuery(array<FetchState.nextQuery>)
+
 /**
 Gets the next query from the fetchState with the lowest latestFetchedBlock number.
 */
-let getNextQueriesOrThrow = (
-  self: t,
-  ~currentBlockHeight,
-  ~maxPerChainQueueSize,
-  ~partitionsCurrentlyFetching,
-) => {
-  let optUpdatedPartition = ref(None)
```

Comment:
Don't check whether there's a change or not. Always update the fetchstate, since it's more expensive to track whether it's changed or not.